crates/router/src/core/payments/transformers.rs

```diff
@@ -601,43 +601,58 @@ where
     let customer_table_response: Option<CustomerDetailsResponse> =
         customer.as_ref().map(ForeignInto::foreign_into);
 
-    // If we have customer data in Payment Intent, We are populating the Retrieve response from the
-    // same
-    let customer_details_response =
-        if let Some(customer_details_raw) = payment_intent.customer_details.clone() {
-            let customer_details_encrypted =
-                serde_json::from_value::<CustomerData>(customer_details_raw.into_inner().expose());
-            if let Ok(customer_details_encrypted_data) = customer_details_encrypted {
-                Some(CustomerDetailsResponse {
-                    id: customer_table_response.and_then(|customer_data| customer_data.id),
-                    name: customer_details_encrypted_data
-                        .name
-                        .or(customer.as_ref().and_then(|customer| {
-                            customer.name.as_ref().map(|name| name.clone().into_inner())
-                        })),
-                    email: customer_details_encrypted_data.email.or(customer
+    // If we have customer data in Payment Intent and if the customer is not deleted, We are populating the Retrieve response from the
+    // same. If the customer is deleted then we use the customer table to populate customer details
+    let payment_intent_customer_details = payment_intent.customer_details.clone();
+    let is_customer_details_redacted = customer_table_response
+        .clone()
+        .and_then(|customer_details| customer_details.name)
+        .as_ref()
+        .map(|customer_name| customer_name.peek().to_owned())
+        .ne(&Some("Redacted".to_owned()));
+
+    let customer_details_response = if payment_intent_customer_details.is_some()
```

Comment:
instead of doing this check, can we have the precedence for the customer details derived from customers table first and then if it is not present, then fallback to the customer details stored in payment intent